docstore/doc.go

```diff
@@ -12,42 +12,42 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-// Package docstore provides a portable implementation of a document store.
-// A document store is a service that stores data in recursively defined
-// documents (like, but not necessarily the same as, JSON). Like other
-// NoSQL databases, document stores are schemaless.
+// Package docstore provides a portable way of interacting with a document store.
+// A document store is a service that stores data in semi-structured JSON-like
+// documents. Like other NoSQL databases, document stores are schemaless.
 // See https://en.wikipedia.org/wiki/Document-oriented_database for more information.
 //
+// Subpackages contain distinct implementations ("drivers") of docstore for public
+// cloud services like Google Cloud Firestore and Amazon DynamoDB as well as any
+// MongoDB service hosted by a cloud provider or on-premise. For example, you can
+// use Azure Cosmos DB and Amazon DocumentDB via the MongoDB driver. There is also
```

Comment:
I think Cosmos and DocDB are not "service hosted by a cloud provider", I think they are just compatible with MongoDB's API.

Reply:
I reworded a bit. I believe these are services, and they are hosted by a provider, but they are not MongoDB services, strictly speaking, so I called out MongoDB-compatible separately (although this page does not: https://www.mongodb.com/cloud/atlas/compare). Wouldn't our mongo driver work with MongoDB Atlas as well?

Reply:
That's right. MongoDB Atlas is something like that -- I think it's a multi-cloud product built by Mongo that will start mongo instances in the cloud.

As Tuo said, Cosmos and DocumentDB are their own products with their own implementations, and they just talk the mongo protocol.

Reply:
Your rewording LGTM.